test/e2e/operator_test.go

```diff
@@ -1075,6 +1075,88 @@ func TestContainerLogging(t *testing.T) {
 	}
 }
 
+func TestIngressControllerCustomEndpoints(t *testing.T) {
+	// TODO: Change from spec to status after https://bugzilla.redhat.com/show_bug.cgi?id=1850681
+	//       is fixed.
+	platform := infraConfig.Spec.PlatformSpec
+	switch platform.Type {
+	case configv1.AWSPlatformType:
+		if len(platform.AWS.ServiceEndpoints) != 0 {
+			t.Fatalf("custom endpoints detected for infrastructure %s", infraConfig.Name)
+		}
+		route53Endpoint := configv1.AWSServiceEndpoint{
+			Name: "route53",
+			URL:  "https://route53.amazonaws.com",
+		}
+		elbEndpoint := configv1.AWSServiceEndpoint{
+			Name: "elasticloadbalancing",
+			URL:  "https://elasticloadbalancing.us-west-2.amazonaws.com",
+		}
+		taggingEndpoint := configv1.AWSServiceEndpoint{
+			Name: "tagging",
+			URL:  "https://tagging.us-east-1.amazonaws.com",
+		}
+		endpoints := []configv1.AWSServiceEndpoint{route53Endpoint, elbEndpoint, taggingEndpoint}
+		infraConfig.Spec.PlatformSpec.AWS.ServiceEndpoints = endpoints
+		if err := kclient.Update(context.TODO(), &infraConfig); err != nil {
+			t.Logf("failed to update infrastructure config: %v\n", err)
+		}
+		// Wait for infrastructure status to update with custom endpoints.
+		err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			if err := kclient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, &infraConfig); err != nil {
+				t.Logf("failed to get infrastructure config: %v\n", err)
+				return false, nil
+			}
+			// TODO: Change from spec to status after https://bugzilla.redhat.com/show_bug.cgi?id=1850681
```

Comment:
The PR from BZ#1850681 has merged. 
https://github.com/openshift/cluster-config-operator/pull/143
With that in mind, should these TODO's be address in this PR?

Reply:
Yeah I saw that PR merge. I plan to update this PR today.